web/beacon-app/src/hooks/useQueryParams.tsx

```diff
@@ -0,0 +1,11 @@
+import queryString from 'query-string';
+import { useLocation } from 'react-router-dom';
+
+const useQueryParams = () => {
+  const location = useLocation();
+  const params = queryString.parse(location.search); // Parse the query string
+
+  return params;
+};
+
+export default useQueryParams;
```

Comment:
By implementing this abstraction, we can efficiently retrieve query parameters while eliminating the need to duplicate identical code across various sections.